### PR TITLE
fix: use a custom remove function to find the file causing the flakiness

### PR DESCRIPTION
**Pull request by SweeXordious**
## Checklist

- [ ] New and updated code has appropriate documentation
- [ ] New and updated code has new and/or updated testing
- [ ] Required CI checks are passing
- [ ] Visual proof for any user facing features like CLI or documentation updates
- [ ] Linked issues closed with keywords

### CODE REVIEW

File: testutil/testnode/full_node.go [hunk 1]
```diff
@@ -214,12 +215,34 @@ func DefaultNetwork(t *testing.T, blockTime time.Duration) (accounts []string, c
...
 		require.NoError(t, stopNode())
 		require.NoError(t, cleanupGRPC())
-		require.NoError(t, os.RemoveAll(tmCfg.RootDir))
+		require.NoError(t, removeDir(path.Join([]string{tmCfg.RootDir, "config"}...)))
 	})
 
 	return accounts, cctx
 }
 
+// removeDir removes the files in the specified directory.
```
Comment (SweeXordious) [inline testutil/testnode/full_node.go:224]:
```suggestion
// removeDir removes the directory `rootDir`.
```

**Review by SweeXordious — COMMENTED** [review]
If we can merge this PR, once we have a flaky CI, we  will know exactly which file is causing this. Then, we can investigate which process is keeping it open even after shutting down the node.

**Review by rootulp — APPROVED** [review]

File: testutil/testnode/full_node.go [hunk 2]
```diff
@@ -214,12 +215,34 @@ func DefaultNetwork(t *testing.T, blockTime time.Duration) (accounts []string, c
...
+		require.NoError(t, removeDir(path.Join([]string{tmCfg.RootDir, "config"}...)))
 	})
 
 	return accounts, cctx
 }
 
+// removeDir removes the directory `rootDir`.
+// The main reason for using it is to know if some file is used by some leaking process during
+// cleanup and be able to identify where the leak is occurring.
+// TODO: remove after fixing the CI flakiness
```
Comment (evan-forbes) [inline testutil/testnode/full_node.go:227]: >TODO: remove after fixing the CI flakiness

does this fix some of the flakiness or is this just a hacky fix?

**Review by evan-forbes — APPROVED** [review]

File: testutil/testnode/full_node.go (hunk 2 above)
Comment (SweeXordious) [reply testutil/testnode/full_node.go:227]: Its a way to know exactly which file is causing the flakiness to investigate which process is leaking. Next time the flakiness happens, this should give us more informatjon (might even fix the flakiness, but not sure)

**Pull request merged**